gpflow/kernels.py

```diff
@@ -232,7 +232,11 @@ def __init__(self, input_dim, variance=1.0, lengthscales=None,
             self.ARD = False
 
     @params_as_tensors
-    def square_dist(self, X, X2):
+    def scaled_square_dist(self, X, X2):
+        """
+        Returns ((X - X2ᵀ)/lengthscales)².
+        Due to the implementation, the result may actually be negative.
+        """
```

Comment:
I'm against renaming this method. Second opinion of @jameshensman? If James agrees to the renaming, I would like to insist on deprecating `square_dist()` with a warning for the time being.

Also, with "the result may be actually be negative", you mean due to floating point precision, right? Could you clarify this in the docstring. Right now it just seems like the implementation is wrong.

Reply:
Happy to add deprecated methods.